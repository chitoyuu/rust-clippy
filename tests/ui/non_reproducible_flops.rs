#![warn(clippy::non_reproducible_flops)]

fn sin(f: f32) -> f32 {
    unimplemented!()
}

fn main() {
    1.0_f32.sin();
    f32::sin(1.0);
    sin(1.0);

    1.0_f32.clamp(2.0, 3.0);
    f32::clamp(1.0, 2.0_f32.hypot(3.0), f32::default());

    f32::from(1u8);

    1.0_f64.abs().mul_add(2.0, 4.0).floor().ln();
}
