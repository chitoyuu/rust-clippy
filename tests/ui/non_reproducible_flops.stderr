error: non-reproducible float operation detected
  --> $DIR/non_reproducible_flops.rs:8:13
   |
LL |     1.0_f32.sin();
   |             ^^^
   |
   = note: `-D clippy::non-reproducible-flops` implied by `-D warnings`

error: non-reproducible float operation detected
  --> $DIR/non_reproducible_flops.rs:9:5
   |
LL |     f32::sin(1.0);
   |     ^^^^^^^^

error: non-reproducible float operation detected
  --> $DIR/non_reproducible_flops.rs:13:29
   |
LL |     f32::clamp(1.0, 2.0_f32.hypot(3.0), f32::default());
   |                             ^^^^^

error: non-reproducible float operation detected
  --> $DIR/non_reproducible_flops.rs:17:45
   |
LL |     1.0_f64.abs().mul_add(2.0, 4.0).floor().ln();
   |                                             ^^

error: aborting due to 4 previous errors

